### ISSUE TITLE
Alternating multiply referred footnotes produce a ? in pdf output

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#10188]
### Describe the bug

In some circumstances footnote mark is rendered as `?` and there is no hyperlink

### How to Reproduce

file `index.rst`: 

```
Test
====

Explicitly numbered footnotes
-----------------------------

First reference to first footnote [1]_ 

First reference to second footnote [2]_

Second reference to first footnote [1]_

Second reference to second footnote [2]_


.. rubric:: Footnotes

.. [1] A first footnote

.. [2] A second footnote
```

then `make latexpdf`.

### Expected behavior

Footnotes are rendered correctly

### Your project

See above code

### Screenshots

![Capture d’écran 2022-02-13 à 09 32 11](https://user-images.githubusercontent.com/2589111/153745645-840efe61-7bdc-4855-99bc-1862f415932a.png)

### OS

Mac

### Python version

3.8.7 (CPython)

### Sphinx version

4.4.0 and current 4.x (v4.5.0+/4ba056870)

**Comment by jfbu** [comment]
The #8832 mechanism fixed some problems with explicitly numbered footnotes (and #10169 extended this to named footnotes). But this mechanism is flawed when new named or explicitly numbered footnotes occur in-between multiple references to same original one.

**Comment by Jellby** [comment]
Is there any workaround for now?

**Comment by Jellby** [comment]
> Is there any workaround for now?

Just `\let\sphinxstepexplicit\relax` does it.

Would it be possible to make LaTeX issue a warning whenever a `?` is used instead of a proper footnote mark? That would save some hunting in the final PDF files.

**Comment by jfbu** [comment]
@Jellby I have now merged for 5.0.0 release the #10191 pull request after some rebasing work. Nice suggestion about letting LaTeX issue a warning in place of a `?`, I will keep it in mind for improving.
